worksheet.rb

```diff
@@ -23,12 +56,98 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
-########################################################
+#######################################################
 # Step 4: Total Driver's Earnings and Number of Rides
-
 # Use an iteration blocks to print the following answers:
+
+def line
+  puts "#######################################################"
+end
+
+line
+
 # - the number of rides each driver has given
+
+def num_ride(ride_share)
+  ride_share.each do |key, value|
+    puts "#{key} has #{value.length} rides."
+    end
+end
+
+num_ride(ride_share)
+
+line
+
 # - the total amount of money each driver has made
+
+def driver_earn(ride_share)
+  ride_share.each do | driver, array |
+    total = 0
+     array.each do |ride|
+       total += ride[:cost]
+    end
+    puts "#{driver} made $#{total}"
+  end
+end
+
+driver_earn(ride_share)
+
+line
+
 # - the average rating for each driver
+
+def driver_avg(ride_share)
+  ride_share.each do | driver, array |
+    total = 0
+     array.each do |ride|
+       total += ride[:rating]
+     end
+    average = total.to_f / array.length
+    puts "#{driver} made #{average.round(1)}"
```

Comment:
Consider returning the average from this method, and then using driver code to puts this information to the user.